### PR TITLE
[HCSR501 ] Update sensor publication criteria

### PR DESCRIPTION
**Pull request by diepeterpan**
## Description:
Improve performance by avoiding unnecessary calls to handleJsonEnqueue when the HCSR501 sensor's state did not change

## Checklist:
  - [x] The pull request is done against the latest development branch
  - [x] Only one feature/fix was added per PR and the code change compiles without warnings
  - [x] I accept the [DCO](https://github.com/1technophile/OpenMQTTGateway/blob/development/docs/participate/development.md#developer-certificate-of-origin).

### CODE REVIEW

File: main/ZsensorHCSR501.ino [hunk 1]
```diff
@@ -65,7 +65,8 @@ void MeasureHCSR501() {
     digitalWrite(HCSR501_LED_NOTIFY_GPIO, pirState == HCSR501_LED_ON);
 #  endif
     HCSR501data["origin"] = subjectHCSR501toMQTT;
-    handleJsonEnqueue(HCSR501data);
+    if (HCSR501data.size() > 1)
+      handleJsonEnqueue(HCSR501data);
```
Comment (1technophile) [inline main/ZsensorHCSR501.ino]: `HCSR501data["origin"] = subjectHCSR501toMQTT;` could be moved inside the condition as it does not need to be done if the size = 0

Comment (diepeterpan) [reply main/ZsensorHCSR501.ino]: Agree, thanks, done :-)

**Comment by 1technophile** [comment]
Thanks

**Comment by diepeterpan** [comment]
Wait a minute, what did I do here, this is not 100% correct, now my one sensor is not working, I am going to rework this, sorrrrrreeeeeee.   Will create a new pull request once I fixed it.

**Pull request merged**